charts/agent-operator/templates/operator-clusterrole.yaml

```
@@ -52,4 +52,11 @@ rules:
   - daemonsets
   - deployments
   verbs: [get, list, watch, create, update, patch, delete]
+{{- with .Values.rbac.podSecurityPolicyName }}
+- apiGroups: [policy]
+  resources:
+  - podsecuritypolicies
+  verbs:         [use]
+  resourceNames: [ {{ . }} ]
+{{- end -}}
```

Do we need to take care k8s 1.25 psp deprecation?

```suggestion
{{- with .Values.rbac.podSecurityPolicyName }}
{{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
- apiGroups: [policy]
  resources:
  - podsecuritypolicies
  verbs:         [use]
  resourceNames: [ {{ . }} ]
{{- end }}
{{- end -}}
```

---

My opinion so far has been the I would leave that to the end user. If they are still configuring a PSP maybe they rely on it; then the chart will fail to install. They can be explicit to not use a PSP and the chart will install ok.

---

I'm not feeling super strongly about this though